testdata/templates/bad-arg.yaml

```diff
@@ -0,0 +1,3 @@
+service: foo
+request:
+  name: ${user:foo
```

Comment:
We cover a lot of the edge cases such as `\\` and `\$` in `TestProcessString` and `TestProcessMap` already. but added a couple more to `args.yaml` -- it has a fallback which is not a string, a list replacement, and testing for escapes.